File: server/graph/tools/dataLoader.js
const fs = require("fs");
const path = require("path");

function readJsonIfExists(filePath, fallback) {
  if (!fs.existsSync(filePath)) {
    return fallback;
  }

  const raw = fs.readFileSync(filePath, "utf-8");
  return JSON.parse(raw);
}

function loadProducts() {
  return readJsonIfExists(
    path.join(__dirname, "../data/products.json"),
    []
  );
}

function loadPlaces() {
  return readJsonIfExists(
    path.join(__dirname, "../data/places.json"),
    []
  );
}

module.exports = {
  loadProducts,
  loadPlaces,
};

File: pygraph/tools/data_loader.py
import json
from pathlib import Path


def load_products():
    path = Path("data/products.json")
    with open(path, "r", encoding="utf-8") as f:
        return json.load(f)


def load_places():
    path = Path("data/places.json")
    if not path.exists():
        return []
    with open(path, "r", encoding="utf-8") as f:
        return json.load(f)